server/amp/helpers/infinite-scroll.js

```diff
@@ -41,34 +50,42 @@ class InfiniteScrollAmp {
     return `${hostWithProtocol}/${s3Key}?format=webp&w=250`;
   }
 
+  async getInfiniteScrollList(storyId) {
+    const sourceType = this.infiniteScroll && this.infiniteScroll.source;
+    let filteredItems = [];
+    if (sourceType === "relatedStoriesApi") {
+      const relatedStoriesList = await this.story.getRelatedStories(this.client);
+      if (!relatedStoriesList)
+        return false;
+      return filteredItems = this.getFilteredApiItems(relatedStoriesList);
+    } else {
+      const collection = await this.client.getCollectionBySlug("amp-infinite-scroll");
+      if (!collection || (collection.items && !collection.items.length) || collection.error || collection === null)
+        return false;
+      return filteredItems = this.getFilteredCollItems(collection, storyId);
+    }
+  }
+
   async getResponse({ itemsTaken }) {
     const { "story-id": storyId } = this.queryParams;
     if (!storyId) return new Error(`Query param "story-id" missing`);
-
-    const collection = await this.client.getCollectionBySlug(this.collSlug);
-    if (!collection || collection.error)
-      return new Error(`Infinite scroll collection ${this.collSlug} returned falsy value`);
-    const filteredItems = this.getFilteredCollItems(collection, storyId);
+    const filteredItems = await this.getInfiniteScrollList(storyId);
+    if (typeof filteredItems === "boolean") return new Error(`Infinite scroll collection amp-infinite-scroll returned falsy value`);
     const slicedItems = filteredItems.slice(itemsTaken);
     const formattedData = this.formatData({ itemsArr: slicedItems });
-
     return JSON.stringify(formattedData);
   }
 
   async getInitialInlineConfig({ itemsToTake, storyId }) {
     if (!itemsToTake || !storyId) return new Error("Required params for getInitialInlineConfig missing");
-
-    const collection = await this.client.getCollectionBySlug(this.collSlug);
-    if (!collection || (collection.items && !collection.items.length) || collection.error) return null;
-    const filteredItems = this.getFilteredCollItems(collection, storyId);
+    const filteredItems = await this.getInfiniteScrollList(storyId);
```

Comment:
if the source is `relatedStoriesApi`, `getInitialInlineConfig` should call the related stories api with offset of 0 and limit of 5